### ISSUE TITLE
Image path is now case sensitive 

### ISSUE THREAD

**Issue by ak-ldube** [doxygen/doxygen#8129]
**Describe the bug**
Image path is now case sensitive as of 1.8.20. Doxygen version 1.8.17 was fine.
`warning: image file images/WhatsNew-mec.PNG is not found in IMAGE_PATH: assuming external image.`

**Expected behavior**
Image path should not be case sensitive

**Screenshots**
![image](https://user-images.githubusercontent.com/72398301/97115112-216e9700-16cb-11eb-9fc7-af0a343371ef.png)

**To Reproduce**
Use doxygen 1.8.20 and just change the case of any image, it can be filename or extension name. I didn't check if case in IMAGE_PATH change anything

**Version**
1.8.20 Windows 10, 64 bit.

**Comment by albert-github** [comment]
What doxygen command / construct did you use to incorporate the image into your documentation?

- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by ak-ldube** [comment]
**\image html images/WhatsNew-mec.PNG**

**Comment by albert-github** [comment]
I did a quick test but my tests gave the expected results / warnings for the 1.8.17 and 1.8.20 version, so:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
I did some further tests and saw that the problem is indeed reproducible. 
Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/5435665/example.tar.gz)

My first guess is that the problem is introduced in:
```
Commit: 4bc2355373979726c7ed4e8351639123daf808cb [4bc2355]
Date: Monday, April 6, 2020 7:19:07 PM
Replaced FileNameDict/FileNameList by FileNameLinkedMap
```
in the 1.8.17 version we have a `filename.cpp` where we see:
```
static bool getCaseSenseNames()
{
  static bool caseSenseNames = Config_getBool(CASE_SENSE_NAMES);
  return caseSenseNames;
}

FileNameDict::FileNameDict(uint size) : QDict<FileName>(size,getCaseSenseNames())
{
}
```
so the dictionary is case sensitive depending on whether this is set in the doxygen settings (for Windows the default value is not case sensitive).

In the 1.8.18 version there is no trace of this case sensitive behavior for file names.

Note: it always better not to use the same case in the `\image` command and in the name of the image (will overcome possible problems when creating the documentation on Linux and Windows).

**Comment by ak-ldube** [comment]
but... but... my examples are ready :(

**Comment by albert-github** [comment]
Still worthwhile to post as maybe I did overlook another corner case.

**Comment by ak-ldube** [comment]
[fails.zip](https://github.com/doxygen/doxygen/files/5435689/fails.zip)
[works.zip](https://github.com/doxygen/doxygen/files/5435690/works.zip)

**Comment by albert-github** [comment]
In the code the I see constructs like:
```
class FileNameLinkedMap : public LinkedMap<FileName>
```
and also:
```
class MemberNameLinkedMap : public LinkedMap<MemberName>
class MemberNameInfoLinkedMap : public LinkedMap<MemberNameInfo>
class RefListManager : public LinkedMap<RefList>
class SectionManager : public LinkedMap<SectionInfo>
```
In the past the `FileNameLinkedMap` was case sensitive / insensitive depending on the setting of `CASE_SENSE_NAMES` at the moment this is not the case. The other maps should be case sensitive.

According to the documentation:
> If the `CASE_SENSE_NAMES` tag is set to `NO` then doxygen will only generate file names in lower-case letters. If set to `YES`, upper-case letters are also allowed. This is useful if you have classes or files whose names only differ in case and if your file system supports case sensitive file names. Windows (including Cygwin) and Mac users are advised to set this option to `NO`.

This signals that this tag is only for output files and not for input files and thus that the implementation in 1.8.17 was a bug.
On the other hand my personal opinion is that it is useful that also input files adhere to this tag. @doxygen ?

I had a look at the possibilities how to handle it as it is possible to give a 3rd argument to the:
```
using Map = std::unordered_map<std::string,T*>;
```
in `linkedmap.h` being the comparison function, but I didn't find a good and easy way to do this, so that the `FileNameLinkedMap` is depending on `CASE_SENSE_NAMES` and the other maps are not (even not after I googled for "cpp unordered_map case insensitive").

**Comment by doxygen** [comment]
I've added an option to make the container class store the keys in lower case, which should restore the original behavior for files.

**Comment by albert-github** [comment]
What will happen on a case sensitive system where there are file with the "same" name but only different in case ? Won't they be mapped to the same name?

**Comment by doxygen** [comment]
@albert-github I think you are right; Looking more closely at the original implementation (based on QDict) if case sensitivity is disabled, I see it the QDict container still stores all keys and then does a case insensitive match when searching (and then probably return the wrong instance) but it does return all keys when iterating.

The current approach will still work fine if CASE_SENSE_NAMES is set to YES in this case, but when disabled it is different from what it was before.

**Comment by albert-github** [comment]
I'm a bit confused by the name of the variable  `case_sensitivity_support` in case of file names this is set to `true` (and in the code later on combined with `Config_getBool(CASE_SENSE_NAMES)`) but when set the used names are possibly case insensitive aren't they (though the usage looks further OK)?

**Comment by doxygen** [comment]
I've solved it differently now, which should be more in line with the original QDict based implementation, i.e. all keys are stored using a multimap but searching and hashing is done with lower case keys.

**Comment by albert-github** [comment]
I think this looks OK, though shouldn't the documentation of `CASE_SENSE_NAMES` not be adjusted as well (saying that `CASE_SENSE_NAMES` isn't only valid for output file names but also for input file names?

**Comment by doxygen** [comment]
@albert-github I agree the documentation should be more clearly state that the `CASE_SENSE_NAMES` option should match the capabilities of the underlying file system. In case the files system supports files whose name only differs in the casing used, the option must be enabled to properly deal with such input files if they appear in the input, whereas for file systems that are not case sensitive the option needs to be disabled to properly deal with output files for symbols that only differ in case, like two classes one named `CLASS` and the other named `Class` and to support references to files without having to specify the correct casing, like was the case for this example.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
